tests/recognizer-tests.js

```diff
@@ -20,6 +20,14 @@ test("A simple route recognizes", function() {
   equal(router.recognize("/foo/baz"), null);
 });
 
+test("A route with URL-encoded parts recognizes", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo bar?baz", handler: handler }]);
```

Comment:
I'd prefer to use another character besides `?` or `#` (how about `$`?)

In fact we should be asserting that these two character do not appear in a path (we may want to reserve them for some DSL later on) as they have very explicit URL parsing behaviors.